integration/tests/cook/test_basic.py

```diff
@@ -24,15 +25,33 @@ def setUp(self):
         util.wait_for_cook(self.cook_url)
 
     def get_job(self, job_uuid):
-        """Loads a job by UUID using GET /rawscheduler"""
-        return util.query_jobs(self.cook_url, job=[job_uuid]).json()[0]
+        """Loads a job by UUID"""
+        return util.get_job(self.cook_url, job_uuid)
+
+    @retry(stop_max_delay=120000, wait_fixed=5000)
+    def get_output_url(self, job_uuid):
+        """
+        Gets the output_url for the given job, retrying every 5 
+        seconds for a maximum of 2 minutes. The retries are 
+        necessary because currently the Mesos agent sandbox
+        directories are cached in Cook.
+        """
+        job = self.get_job(job_uuid)
+        instance = job['instances'][0]
+        if 'output_url' in instance:
+            return instance['output_url']
+        else:
+            error_msg = 'Job %s had no output_url' % job['uuid']
+            self.logger.info(error_msg)
+            raise RuntimeError(error_msg)
```

Comment:
Generic comment: should we use a version of `retry` that retries when `None` (e.g. `@retry(retry_on_result=retry_if_result_none)`) is the result instead of generating and throwing exceptions?